token/ts/src/actions/getOrCreateAssociatedTokenAccount.ts

```diff
@@ -0,0 +1,86 @@
+import {
+    Commitment,
+    ConfirmOptions,
+    Connection,
+    PublicKey,
+    sendAndConfirmTransaction,
+    Signer,
+    Transaction,
+} from '@solana/web3.js';
+import { ASSOCIATED_TOKEN_PROGRAM_ID, TOKEN_PROGRAM_ID } from '../constants';
+import {
+    TokenAccountNotFoundError,
+    TokenInvalidAccountOwnerError,
+    TokenInvalidMintError,
+    TokenInvalidOwnerError,
+} from '../errors';
+import { createAssociatedTokenAccountInstruction } from '../instructions';
+import { Account, getAccountInfo, getAssociatedTokenAddress } from '../state';
+
+/**
+ * Retrieve the associated token account, or create it if it doesn't exist
+ *
+ * @param connection               Connection to use
+ * @param payer                    Payer of the transaction and initialization fees
+ * @param mint                     Mint associated with the account to set or verify
+ * @param owner                    Owner of the account to set or verify
+ * @param commitment               Desired level of commitment for querying the state
+ * @param confirmOptions           Options for confirming the transaction
+ * @param programId                SPL Token program account
+ * @param associatedTokenProgramId SPL Associated Token program account
+ *
+ * @return Address of the new associated token account
+ */
+export async function getOrCreateAssociatedTokenAccount(
```

Comment:
Is there a reason why the parameter `allowOwnerOffCurve` is not exposed here, and hard coded to `false` in the call to `getAssociatedTokenAddress`? Would like to use this function to create an associated token account for a PDA.

Reply:
I think we could expose it, creating an ATA for a PDA is probably a common thing. @joncinque wdyt?

Reply:
Added in 90e2aecfdb869779a4c0948e261a0cc4eff28d85

Reply:
Yep, that makes a lot of sense!